kong/plugins/zipkin/handler.lua

```diff
@@ -136,6 +136,13 @@ if subsystem == "http" then
     -- Want to send headers to upstream
     local proxy_span = zipkin.proxy_span
     local set_header = kong.service.request.set_header
+
+    -- Set the W3C Trace Context header
+    -- TODO: Should W3C traceparent header be set even it wasn't on the incoming request?
```

Comment:
I asked myself a very similar question when implementing the b3 single header (#66).

In order to minimize the amount of configuration involved, I decided to take a salomonical approach: The plugin *understands* b3 single headers, but it only produces "regular" headers (such as `x-b3-traceid`).

It seemed an ok solution for the time but since we are adding a third kind of header I think we should control it via a new config file. A possible design would be:

`options.header_types` is a new config option with 3 possible values:
* `preserve`: the plugin produces whatever format it receives on each request
* `b3-single`: always use b3 single
* `b3`: always use the "regular zipkin" headers, such as `x-b3-traceid`
* `w3c-trace-context`: use the format proposed on this PR

When `options.header_types` is set to anything other than `preserve`, and the trace headers don't match (for example the config is b3-single but we receive a w3c-trace-context) then we should send *both* kinds of headers. We should also log a warning about the inconsistency.

If you want me to take over adding this, please let me know.

Reply:
@victorNewRelic, do you intend to continue with this patch?